src/serial-console/azext_serialconsole/custom.py

```diff
@@ -0,0 +1,723 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+
+import json
+import threading
+import sys
+import uuid
+import time
+import re
+import textwrap
+import websocket
+import requests
+from azure.cli.core.azclierror import UnclassifiedUserFault
+from azure.cli.core.azclierror import ResourceNotFoundError
+from azure.cli.core.azclierror import AzureConnectionError
+from azure.cli.core.azclierror import ForbiddenError
+from azure.cli.core.commands.client_factory import get_subscription_id
+from azure.cli.core._profile import Profile
+from azure.core.exceptions import ResourceNotFoundError as ComputeClientResourceNotFoundError
+from azext_serialconsole._client_factory import _compute_client_factory
+
+
+ARM_ENDPOINT = "https://management.azure.com"
+RP_PROVIDER = "Microsoft.SerialConsole"
```

Comment:
ARM ENDPOINT will be changed when the could is switched. So this value should be read from `cli_ctx.cloud` property. It's defined here [https://github.com/Azure/azure-cli/blob/3feea02888ea67f033f407174a3a7a340158b81a/src/azure-cli-core/azure/cli/core/cloud.py#L297-L332](https://github.com/Azure/azure-cli/blob/3feea02888ea67f033f407174a3a7a340158b81a/src/azure-cli-core/azure/cli/core/cloud.py#L297-L332)